src/ui/SearchInterface/FacetValueStateHandler.ts

```diff
@@ -0,0 +1,50 @@
+import { QueryStateModel } from '../../models/QueryStateModel';
+import { Component } from '../Base/Component';
+import { BaseComponent } from '../Base/BaseComponent';
+
+export class FacetValueStateHandler {
+  constructor(private componentsFetcher: (componentId: string) => Component[]) {}
+
+  public handleFacetValueState(stateToSet: { [key: string]: any }): void {
```

Comment:
I think it would greatly help readability if we could break this method into two or more smaller methods having expressive names describing what they do. Maybe the logic in the first `if` block could be encapsulated in a method called `getfieldsWithoutFacets`, the second `transformFieldsIntoHiddenQueries`?

I have joined @dms1lva 's philosophy school which says:

> if you find yourself needing to leave comments, it is likely that the readability of the code can be improved by using variables/method names

:P